contributors/design-proposals/arbitrator.md

```diff
@@ -0,0 +1,284 @@
+# Policy based resource sharing of Kubernetes
+
+@k82cn
+
+January, 2017
+
+## User Scenario and Requirements:
+
+As a cluster admin, I’d like to build an environment to run different workload together, e.g. long running service, bigdata.
+ As those applications are managed by different department, I have to provide resource guarantee to each applications, 
+ demonstrated as following:
+
+* Long running service (app area) and bigdata (bigdata area) can share resources:
+  - Define resource usage of each area, e.g. 40% resources to app area, 60% to bigdata area.
+  - Borrow/lending protocol: if the resources is idle in one area, it can be lend out and be preempted back
+* Run multiple cluster in bigdata area:
+  - Define resources usage of each cluster within bigdata area, e.g. Spark, Hadoop
+  - Sharing resources between those big data clusters, e.g. borrow/lending protocol
+
+The detail of requirements for the "bigdata" are
+
+* run a set of applications
+* provided each application guaranteed access to some quantity of resources
+* provided all applications best-effort access to all unused resources according to some target weight (one weight assigned
+  to each application, i.e. if all applications wanted to use all free resources, then they would be allowed to do so 
+  in some relative proportion)
+* If some application A is using less than its guarantee, and then if it decides to use its guarantee and there aren't 
+  enough free resources to do so, it should be able to evict tasks from some other application or applications
+  (that is/are using more than their guarantee) in order to obtain its guarantee
+
+Further, group "bigdata" apps and "service" apps into two buckets, providing each bucket (in aggregate) guaranteed 
+access to some fraction of the cluster, and best-effort access to the entire cluster with the understanding that usage 
+above the guarantee can be revoked at any time. 
+
+According to this [mesos-style.md](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/mesos-style.md), 
+it definitely can build a customized components to do so; but it’s better for Kubernetes to provide such a general feature
+ about resource planning, resource management and resource sharing.
+
+## Terminology
+
+* Arbitrator: A new components/binary that allocate resources based on policy (resource allocation).
+  The default scheduler still assign Pods to resources (resource assignment) 
+* Deserved (Resource): The total number of resources that the arbitrator allocated to the namespace
+* Overused: The namespace is overused if it used more resources than deserved
+
+## Context
+
+As Kubernetes growing up, there’s several features on resource QoS and sharing between namespaces.
+
+### Admission and ResourceQuota:
```

Comment:
I am not sure if discussion of ResourceQuota helps or hurts this proposal as it never asserted that it could support a priority/preemption scheme.   Do you think the proposal would be stronger to remove it and instead just focus on arbitration scheme when pods are unable to be scheduled?